### PR TITLE
Document async and inetaddress DNS

### LINKED ISSUE #25307: Document how (and why) to enable async-dns (and its capabilities)
This is a follow up to - Support SRV records in DNS lookups #25173

We should document it, even if main users of it is "us" for now.
Esp since we tell people to enable it in akka-management so it makes sense to explain what people are actually enabling there.

cc @chbatey

### PR DESCRIPTION
**Pull request by chbatey**
Closed #25307

### CODE REVIEW

File: akka-docs/src/main/paradox/io-dns.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+:  @@snip [IODocSpec.scala]($code$/scala/docs/actor/io/dns/DnsCompileOnlyDocSpec.scala) { #actor-api-async }
+
+Java
+:  @@snip [DnsCompileOnlyDocTest.java]($code$/java/jdocs/actor/io/dns/DnsCompileOnlyDocTest.java) { #actor-api-async }
+
+The Async DNS provider has the following advantages:
+
+* No JVM DNS caching. It is expected that future versions will expose more caching related information.
+* No blocking. `InetAddress` resolving is a blocking operation.
+* Exposes `SRV`, `A` and `AAAA` records
```
Comment (johanandren) [inline akka-docs/src/main/paradox/io-dns.md]: First two ends with a period but last bullet ends without period

File: akka-docs/src/test/java/jdocs/actor/io/dns/DnsCompileOnlyDocTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        //#resolve
+        Option<Dns.Resolved> initial = Dns.get(system).cache().resolve("google.com", system, actorRef);
+        Option<Dns.Resolved> cached = Dns.get(system).cache().cached("google.com");
+        //#resolve
+
+        {
+            //#actor-api-inet-address
+            final ActorRef dnsManager = Dns.get(system).manager();
+            ask(dnsManager, new Dns.Resolve("google.com"), timeout);
```
Comment (johanandren) [inline akka-docs/src/test/java/jdocs/actor/io/dns/DnsCompileOnlyDocTest.java]: Just asking and not doing anything with the returned CS seems strange. Capture in a typed result or maybe add some side effect println or something just to show that we get something back?

**Review by johanandren — COMMENTED** [review]
Lookin' good. Added one nitpick and one actual comment.

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-actor/src/main/scala/akka/io/dns/DnsProtocol.scala [hunk 3]
```diff
@@ -48,17 +48,17 @@ object DnsProtocol {
...
   object Resolve {
     def apply(name: String): Resolve = Resolve(name, Ip())
+  }
 
-    /**
-     * Java API
-     */
-    def create(name: String): Resolve = Resolve(name, Ip())
+  /**
+   * Java API
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/io/dns/DnsProtocol.scala]: actual docs here please too ;-)

File: akka-docs/src/main/paradox/io-dns.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+Beware that the `async-dns` API is marked as `ApiMayChange` as more information is expected to be added.
```
Comment (ktoso) [inline akka-docs/src/main/paradox/io-dns.md]: this should be a `.. note` in the beginning of the page

File: akka-docs/src/test/java/jdocs/actor/io/dns/DnsCompileOnlyDocTest.java (hunk 2 above)
Comment (ktoso) [reply akka-docs/src/test/java/jdocs/actor/io/dns/DnsCompileOnlyDocTest.java]: Agreed :)

**Review by ktoso — APPROVED** [review]
Minor things to address, otherwise looking good!

**Comment by patriknw** [comment]
@chbatey two mini mini nitpicks here and then we can reduce the open PRs with one :)

**Comment by chbatey** [comment]
Updated. Can merge after validation.

**Pull request merged**